src/coreclr/jit/emit.h

```diff
@@ -2129,6 +2140,10 @@ class emitter
     void emitEnableGC();
 #endif // !defined(JIT32_GCENCODER)
 
+#if defined(TARGET_XARCH)
+    bool emitInstHasNoCode(instrDesc* id);
```

Comment:
Why is this needed? Is it because of addition of `instrDesc` parameter?

Reply:
The function is only used in xarch builds and yes it was a knockon effect of needing separate function definitions for xarch and other builds.